crates/cheatcodes/src/evm/fork.rs

```diff
@@ -385,6 +385,16 @@ fn rpc_call(url: &str, method: &str, params: &str) -> Result {
             DynSolValue::Bytes(bytes.as_slice()[..size].to_vec())
         }
         DynSolValue::Address(addr) => DynSolValue::Bytes(addr.to_vec()),
+        DynSolValue::Tuple(mut vals) => {
+            vals.iter_mut().for_each(|val| {
+                // Convert fixed bytes to bytes in tuple to prevent encoding issues.
+                // See: <https://github.com/foundry-rs/foundry/issues/7858>
+                if let DynSolValue::FixedBytes(bytes, size) = *val {
+                    *val = DynSolValue::Bytes(bytes.as_slice()[..size].to_vec());
```

Comment:
pls just extract the function and make it recursive

Reply:
changed in https://github.com/foundry-rs/foundry/pull/9117/commits/f38f5784e30451bf0c647bf431f4d0fc61e83d11, please check